Cereal/CerealDecoder.swift

```diff
@@ -238,14 +275,14 @@ public struct CerealDecoder {
     - returns:      The instantiated object, or nil if no object was at the specified key.
     */
     public func decodeCereal<DecodedKeyType: protocol<CerealType, Hashable>, DecodedValueType: CerealRepresentable>(key: String) throws -> [[DecodedKeyType: DecodedValueType]]? {
-        guard let data = items[key] else {
+        guard let data = self.itemForKey(key), case let .ArrayValue(items) = data else {
```

Comment:
Something like `CerealError.TypeMismatch`. It'd be helpful if someone attempted to decode a value to a type it wasn't support to be; nil means maybe it wasn't ever encoded, but an error means they are decoding (or encoded it) incorrectly